### PR TITLE
Ramp Checks

### LINKED ISSUE #50: Time step consistency between ramping rates and time series

### PR DESCRIPTION
**Pull request by sourabhdalvi**
Added Ramp Checks 
Fixed polynomial cost functions
Added default Ramp rates in constructors
Added New scalingfactor with more than one points to find time-delta for ramp-check
Fixes #50

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -14,9 +14,10 @@ const DATA_FOLDER = Pkg.dir("PowerSystems/data")
 function download_data()
 
     if !isdir(DATA_FOLDER)
-        mkdir(DATA_FOLDER)
-        temp_folder = mktempdir()
-        temp_data_zip = joinpath(temp_folder, "data.zip")
+        mkpath(DATA_FOLDER)
```
Comment (jd-lara) [inline deps/build.jl]: This is making the Windows check fail, why did you change it back? Please remove this commit.

File: src/PowerSystems.jl [hunk 2]
```diff
@@ -71,6 +71,7 @@ using PowerModels
 using DataFrames
 using CSV
 using Memento
+using StatsBase
```
Comment (jd-lara) [inline src/PowerSystems.jl]: What are you using StatsBase for?

File: src/models/generation/hydro_generation.jl [hunk 3]
```diff
@@ -23,7 +23,7 @@ TechHydro(;installedcapacity = 0.0,
           realpowerlimits = @NT(min = 0.0, max = 0.0),
           reactivepower = nothing,
           reactivepowerlimits = nothing,
-          ramplimits = nothing,
+          ramplimits = @NT(up=0.0, down=0.0),
```
Comment (jd-lara) [inline src/models/generation/hydro_generation.jl]: Why are you adding default 0.0 ramps?

File: src/models/generation/thermal_generation.jl [hunk 4]
```diff
@@ -139,6 +139,6 @@ end
 ThermalGenSeason(; name = "init",
                 status = false,
                 bus = Bus(),
-                tech = nothing,
-                econ = nothing,
-                scalingfactor = TimeSeries.TimeArray(today(), [1.0])) = ThermalGenSeason(name, status, bus, tech, econ, scalingfactor)
+                tech = TechThermal(),
+                econ = EconThermal(),
+                scalingfactor = TimeArray(collect(DateTime(today()):Hour(1):DateTime(today()+Day(1))), ones(25))) = ThermalGenSeason(name, status, bus, tech, econ, scalingfactor)
```
Comment (jd-lara) [inline src/models/generation/thermal_generation.jl]: This default scaling factor creates 24 steps by default, there is ni need for that.

Comment (sourabhdalvi) [reply src/models/generation/thermal_generation.jl]: I needed more than 1 step to measure the time delta, I can change it to have just 2 steps if you want

File: deps/build.jl (hunk 1 above)
Comment (sourabhdalvi) [reply deps/build.jl]: I didn't make that change

File: src/PowerSystems.jl (hunk 2 above)
Comment (sourabhdalvi) [reply src/PowerSystems.jl]: Not me again

File: src/models/generation/hydro_generation.jl (hunk 3 above)
Comment (sourabhdalvi) [reply src/models/generation/hydro_generation.jl]: I can change to 10.0 , it just needs to be something instead of a Nothing

File: src/models/generation/thermal_generation.jl [hunk 5]
```diff
@@ -56,7 +56,7 @@ TechThermal(; realpower = 0.0,
           realpowerlimits = @NT(min = 0.0, max = 0.0),
           reactivepower = nothing,
           reactivepowerlimits = nothing,
-          ramplimits = @NT(up=0.0, down=0.0),
+          ramplimits = @NT(up=10.0, down=10.0),
```
Comment (jd-lara) [inline src/models/generation/thermal_generation.jl]: The default should be nothing, I still don't get why are you adding values here. The whole point is to make it work when there are not ramping limits as a nothing.

File: src/models/generation/renewable_generation.jl [hunk 6]
```diff
@@ -54,7 +54,7 @@ RenewableCurtailment(; name = "init",
                 bus= Bus(),
                 installedcapacity = 0.0,
                 econ = EconRenewable(),
-                scalingfactor = TimeArray(collect(DateTime(today()):Hour(1):DateTime(today()+Day(1))), ones(25))) = RenewableCurtailment(name, status, bus, installedcapacity, econ, scalingfactor)
+                scalingfactor =  TimeArray([round(now(),Hour(1)),round(now()+Hour(1),Hour(1))],ones(2))) = RenewableCurtailment(name, status, bus, installedcapacity, econ, scalingfactor)
```
Comment (jd-lara) [inline src/models/generation/renewable_generation.jl]: Also, still don't get why you are making the time series in this format.
